test/framework/src/main/java/org/elasticsearch/common/logging/JsonLogsIntegTestCase.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.test.rest.ESRestTestCase;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Tests that extend this class verify that all json layout fields appear in the first few log lines after startup
+ * Fields available from the first log line: type, timestamp, level, component, message, nodeName, clusterName
+ * whereas: nodeId and clusterId are available later once the clusterState was received.
+ *
+ * NodeName, ClusterName, NodeId, ClusterId should not change across all log lines
+ *
+ * Note that this won't pass for nodes in clusters that don't have the node name defined in elasticsearch.yml <strong>and</strong> start
+ * with DEBUG or TRACE level logging. Those nodes log a few lines before the node.name is set by LogConfigurator.setNodeName
+ */
+public abstract class JsonLogsIntegTestCase extends ESRestTestCase {
+    /**
+     * Number of lines in the log file to check for the node_name, node.id or cluster.uuid. We don't
```

Comment:
I think `node_name` should not be mentioned in this comment? Also, I'd prefer to wrap names like `node.id` and `cluster.uuid` in `<code>` tags.

Reply:
this test is still is asserting about the cluster_name, so I think it still would be good to keep it

Reply:
You're right about the node name, I missed that before.